enterprise/server/cmd/ci_runner/main.go

```diff
@@ -149,6 +149,7 @@ var (
 	visibility         = flag.String("visibility", "", "If set, use the specified value for VISIBILITY build metadata for the workflow invocation.")
 	bazelSubCommand    = flag.String("bazel_sub_command", "", "If set, run the bazel command specified by these args and ignore all triggering and configured actions.")
 	recordRunMetadata  = flag.Bool("record_run_metadata", false, "Instead of running a target, extract metadata about it and report it in the build event stream.")
+	timeout            = flag.Duration("timeout", 100*time.Hour, "Timeout before all commands will be canceled automatically.")
```

Comment:
nit: to make a timeout "optional" in other places, we usually make the default `0` and then only apply a ctx timeout if this is nonzero